### DISCUSSION TITLE
Plotting without plotting packages

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#4453] [Q&A] [answered]
I would like to understand how plots are achieved in `typst`. For instance, can we reproduce the plot below _without_ resorting to a plot package like `cetz`?

![image](https://github.com/typst/typst/assets/75945439/dcafac52-c3ff-4abe-bc87-ebc123419579)

**Comment by jamesrswift** [comment]
Absolutely (after all, cetz is written purely on typst). But it won't be a fun task by any means

**Reply by psads-git** [reply]
Thanks for your reply, @JamesxX. So, from what I infer from your answer, to reproduce the plot I have shared will be a Herculean task when using _only_ `typst` for that, right?

**Reply by hunkyjimpjorps** [reply, marked as answer]
"Herculean effort" is probably overselling it, but in order to draw it, you'd essentially be writing your own incomplete version of cetz.

You can see how it's done in [line.typ](https://github.com/cetz-package/cetz/blob/master/src/lib/plot/line.typ) in cetz itself: it's essentially just taking the set of points, transforming them to coordinates appropriate for the size of the plot on the paper, and then drawing lines through them all.  It's not a complicated task, but you'd be reinventing the wheel to do it yourself.

**Reply by psads-git** [reply]
Thanks, @hunkyjimpjorps, for direct me to `line.typ`: that is very interesting!
